addons/udes_sale_stock/models/stock_move.py

```diff
@@ -48,3 +47,25 @@ def _prepare_procurement_values(self):
         values = super()._prepare_procurement_values()
         values["sale_line_id"] = self.sale_line_id.id
         return values
+
+    def _search_picking_for_assignation(self):
+        """
+        When searching for pickings to assign moves too, the picking should have the
+        same priority as the sales order that the move is created from
+        """
+        pickings = super(StockMove, self)._search_picking_for_assignation()
+        if self.sale_line_id:
+            priority = self.sale_line_id.order_id.priority
+            pickings = pickings.filtered(lambda p: p.priority == priority)
+        return pickings
+
+    def _get_new_picking_values(self):
+        """
+        When creating a picking from a move, the picking should have the same priority
+        as sales order from which it is created.
+        """
+        values = super(StockMove, self)._get_new_picking_values()
+        if self.sale_line_id:
+            # Update the values dictionary with the priority of the associated sale order
+            values.update({"priority": self.sale_line_id.order_id.priority})
```

Comment:
From the code it looks like that self can be multi recordset which eventually will raise error when trying to access if moves we are creating pickings are from different sale orders. Dont know if is a real scenario, saying just looking at the code